src/Features/JsonPatch.SystemTextJson/src/Internal/ConversionResultProvider.cs

```diff
@@ -0,0 +1,95 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Text.Json;
+using System.Text.Json.Nodes;
+
+namespace Microsoft.AspNetCore.JsonPatch.SystemTextJson.Internal;
+
+internal static class ConversionResultProvider
+{
+    internal static ConversionResult ConvertTo(object value, Type typeToConvertTo, JsonSerializerOptions serializerOptions)
+    {
+        if (value == null)
+        {
+            return new ConversionResult(IsNullableType(typeToConvertTo), null);
+        }
+
+        if (typeToConvertTo.IsAssignableFrom(value.GetType()))
+        {
+            // No need to convert
+            return new ConversionResult(true, value);
+        }
+
+        // Workaround for the https://github.com/dotnet/runtime/issues/113926
+        if (typeToConvertTo.Name == "JsonValuePrimitive`1")
+        {
+            typeToConvertTo = typeof(JsonNode);
```

Comment:
This should probably match below (`JsonElement`). I think either is fine as long as it's consistent.

Reply:
Refactored to reuse a chunk of common code.